src/main/java/picard/illumina/BarcodeMetric.java

```diff
@@ -0,0 +1,134 @@
+     */
+    public long ONE_MISMATCH_MATCHES = 0;
+    /**
+     * The number of PF reads matching this barcode that matched with 1 error or no-call.
+     */
+    public long PF_ONE_MISMATCH_MATCHES = 0;
+    /**
+     * The fraction of all reads in the lane that matched to this barcode.
+     */
+    public double PCT_MATCHES = 0d;
+    /**
+     * The rate of all reads matching this barcode to all reads matching the most prevelant barcode. For the
+     * most prevelant barcode this will be 1, for all others it will be less than 1 (except for the possible
+     * exception of when there are more orphan reads than for any other barcode, in which case the value
+     * may be arbitrarily large).  One over the lowest number in this column gives you the fold-difference
+     * in representation between barcodes.
+     */
+    public double RATIO_THIS_BARCODE_TO_BEST_BARCODE_PCT = 0d;
+    /**
+     * The fraction of PF reads in the lane that matched to this barcode.
+     */
+    public double PF_PCT_MATCHES = 0d;
+
+    /**
+     * The rate of PF reads matching this barcode to PF reads matching the most prevelant barcode. For the
+     * most prevelant barcode this will be 1, for all others it will be less than 1 (except for the possible
+     * exception of when there are more orphan reads than for any other barcode, in which case the value
+     * may be arbitrarily large).  One over the lowest number in this column gives you the fold-difference
+     * in representation of PF reads between barcodes.
+     */
+    public double PF_RATIO_THIS_BARCODE_TO_BEST_BARCODE_PCT = 0d;
+
+    /**
+     * The "normalized" matches to each barcode. This is calculated as the number of pf reads matching
+     * this barcode over the sum of all pf reads matching any barcode (excluding orphans). If all barcodes
+     * are represented equally this will be 1.
+     */
+    public double PF_NORMALIZED_MATCHES;
+
+    protected byte[][] barcodeBytes;
+
+    public BarcodeMetric(final String barcodeName, final String libraryName,
+                         final String barcodeDisplay, final String[] barcodeSeqs) {
+
+        this.BARCODE = barcodeDisplay;
+        this.BARCODE_WITHOUT_DELIMITER = barcodeDisplay.replaceAll(IlluminaUtil.BARCODE_DELIMITER, "");
+        this.BARCODE_NAME = barcodeName;
+        this.LIBRARY_NAME = libraryName;
+        this.barcodeBytes = new byte[barcodeSeqs.length][];
+        for (int i = 0; i < barcodeSeqs.length; i++) {
+            barcodeBytes[i] = htsjdk.samtools.util.StringUtil.stringToBytes(barcodeSeqs[i]);
+        }
+    }
+
+    /**
+     * This ctor is necessary for when reading metrics from file
+     */
+    public BarcodeMetric() {
+        barcodeBytes = null;
+    }
+
+    /**
+     * Creates a copy of metric initialized with only non-accumulated and non-calculated values set
+     */
+    public static BarcodeMetric copy(final BarcodeMetric metric) {
```

Comment:
Why not just make this non-static so you can call it  directly on instances instead of passing in an instance?